spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -187,9 +198,18 @@ private void doTaskStart() {
 			if(this.applicationArguments != null) {
 				args = Arrays.asList(this.applicationArguments.getSourceArgs());
 			}
-
-			this.taskExecution = this.taskRepository.createTaskExecution(
-					this.taskNameResolver.getTaskName(), new Date(), args);
+			if(this.taskExecutionId != null) {
+				TaskExecution taskExecution = taskExplorer.getTaskExecution(this.taskExecutionId);
+				Assert.notNull(taskExecution, String.format("Invalid TaskExecution, ID %s not found", this.taskExecutionId));
+				Assert.isNull(taskExecution.getEndTime(), String.format(
+						"Invalid TaskExecution, ID %s task is already complete", this.taskExecutionId));
+				this.taskExecution = this.taskRepository.startTaskExecution(this.taskExecutionId,
+						this.taskNameResolver.getTaskName(), new Date(), args, externalExecutionId);
+			}
+			else {
+				this.taskExecution = this.taskRepository.createTaskExecution(
```

Comment:
In 2.0, we should consider passing a `TaskExecution` instead of the itemized parameters to minimize breaking changes like this going forward.  We can always document what we do with it internally.